From 526c715ce729ce0e0b68c7c45b201903277e2484 Mon Sep 17 00:00:00 2001
From: Mike Zeller <mike.zeller@joyent.com>
Date: Wed, 14 Feb 2018 15:50:33 -0800
Subject: [PATCH] TRITON-144 cloudapi is missing a "field" parameter in
 duplicate IP errors     Reviewed by: Trent Mick <trent.mick@joyent.com>    
 Approved by: Trent Mick <trent.mick@joyent.com>

---
 lib/errors.js         | 1 +
 test/machines.test.js | 1 +
 2 files changed, 2 insertions(+)

diff --git a/lib/errors.js b/lib/errors.js
index 9a7c949..e74be82 100644
--- a/lib/errors.js
+++ b/lib/errors.js
@@ -438,6 +438,7 @@ function translateDuplicateIpErrors(error) {
              * have provisioned the zone.
              */
             return {
+                field: 'ip',
                 code: 'UsedBy',
                 message: 'IP in use'
             };
diff --git a/test/machines.test.js b/test/machines.test.js
index 848f19c..67c1380 100644
--- a/test/machines.test.js
+++ b/test/machines.test.js
@@ -1628,6 +1628,7 @@ test('Create Machine using network and in use IP', function (t) {
             code: 'InvalidParameters',
             message: 'Invalid parameters',
             errors: [ {
+                field: 'ip',
                 code: 'UsedBy',
                 message: 'IP in use'
             } ]
-- 
2.21.0

